src/hooks/lib/api/useApiConcentratorPrimaryAssets.ts

```diff
@@ -0,0 +1,40 @@
+import { Address } from "wagmi"
+
+import { useApiConcentratorStaticData } from "@/hooks/lib/api/useApiConcentratorStaticData"
+import { useCategoriesByPrimaryAsset } from "@/hooks/useCategoriesByPrimaryAsset"
+
+export function useApiConcentratorPrimaryAssets({
+  concentratorTargetAssets,
+}: {
+  concentratorTargetAssets: Address[] | undefined
+}) {
+  const filterCategoriesByPrimaryAsset = useCategoriesByPrimaryAsset()
+  const apiQuery = useApiConcentratorStaticData()
+  const targetAssetToPrimaryAsset: Record<Address, Address> = {} // target to primaryKey
+  apiQuery.data?.map((data) => {
```

Comment:
I think we should move towards creating maps like this in the query functions themselves so that the results are cached and don't need to be performed every time we use a hook like this. That being said, I think this is OK for now because we use a similar pattern elsewhere. It can be optimized later.